File: vz03/src/renders/facilities/road/road.cpp
/* NIM/Nama  : 13515AAA / AAA AAA   */
/* File      : Road.cpp             */

#include <iostream>
#include "road.h"
using namespace std;

/***** ROAD *****/
Road::Road() : Facility('-', WHITE)
{}

Road::Road(char _id, Color _color) : Facility(_id, _color)
{}

Road::Road(const Road& R) : Facility(R)
{}

Road::~Road()
{}

Road& Road::operator=(const Road& R)
{
  Facility::operator=(R);
  return *this;
}

char Road::render()
{
	return id;
}

Color Road::getColor()
{
	return color;
}

Road* Road::clone() const {
  return new Road(*this);
}

File: Road.java
/**
 * @author NIM / Nama : 13515087 / Audry Nyonata
 *  Nama file : Road.java
 */

import java.util.*;
import java.lang.*;

public class Road extends Facility{
   /**
   * Constructor
   */
  public Road(){
    super();
  }
  /**
   * Mengembalikan karakter untuk ditampilkan ke layar
   * @return id bertipe char
   */
  public char render(){
    return '-';
  }
  /**
   * Mengembalikan warna untuk ditampilkan ke layar
   * @return color warna renderable
   */
  public String renderWithColor(){
    return ANSI_WHITE + render() + ANSI_RESET;
  }
}
